### PR TITLE
Adding manual option to deposit schedule selection

### LINKED ISSUE #4696: [Manual deposits] Deposit Scheduling: Merchants can select manual schedule.
### Description

#4464 added the ability for merchants to choose an automatic schedule for deposits (daily, weekly, monthly) but to meet the WCPay 4.7 release window the ability for a merchant to select a manual schedule was descoped.

Add a set of controls to the Payments → Settings page which allows the merchant to change their deposit schedule from an automatic schedule to manual.

Deposits section

### Acceptance criteria

- When `Automatic` is selected, the automatic schedule options are available (Daily, Weekly, Monthly)
- When `Manual` is selected the automatic schedule options are disabled (but still visible).
- The "Deposit schedule" controls are implemented as per the designs W8FHz6thdADIqhlPlBo4Mf-fi-1718%3A77943
- When the merchant saves their settings via the "Save changes" button, their deposit schedule is updated in Stripe via WCPay Server as requested.
- The "Learn more" links that appear in the helper text link to https://woocommerce.com/document/payments/faq/change-deposit-schedule/

**Post-release:** Add user docs for this feature to https://woocommerce.com/document/payments/

### Designs

W8FHz6thdADIqhlPlBo4Mf-fi-1718%3A77943

### Testing instructions

TBC

### Additional context

Project thread: pdjTHR-1c8-p2
PRD: pdjTHR-1bY-p2

### PR DESCRIPTION
**Pull request by brucealdridge**
Fixes #4696

#### Changes proposed in this Pull Request

Introduce top-level radio selections, `Automatic` and `Manual`, for deposits schedule.

#### Testing instructions

* Run `npm run build:client`.
* Go to wp-admin > Payments > Settings.
* Make sure `Automatic` and `Manual` radios appear below `Deposit schedule`.
* Make sure they are functional, eg selecting `Manual` and clicking the `Save changes` button will set the Stripe account payout schedule to `Manual` (check the Stripe dashboard to confirm this).
* The deposit settings match the designs → paJDYF-6a4-p2

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by shendy-a8c** [comment]
@brucealdridge mentioned p1662608177560469/1662603893.523189-slack-C03KT8NLL13 that currently, the UI forgets previous selection when switch to manual, that is:
- Select `Automatic deposits`.
- Select `Weekly`.
- Then select `Manual deposits`.
- Then select `Automatic deposits`.
- Notice that the dropdown is now set to `Daily` instead of `Weekly`.

IMO that is fine. To make it *not* forget, we would need to save previous state / unsaved state and I think that is too over-engineering the problem.

Let me know if making the UI remember is a must have.

**Comment by Jinksi** [comment]
> IMO that is fine. To make it _not_ forget, we would need to save previous state / unsaved state and I think that is too over-engineering the problem.

I agree that it is probably not important, since the user is most likely going to select Manual or Automatic, and not flick between like we are in testing.

The Weekly/Monthly anchor persists when toggling automatic/manual and interval options. This makes it slightly less painful if the user accidentally selected "Manual" and can't remember which day of the month they had previously selected.

**Comment by Jinksi** [comment]
Should we be disabling the form/inputs while saving is in progress?
Currently, the form inputs are enabled and can be changed while the settings request is in progress. This could be confusing for users, especially if the request takes a while.

Note that this applies to the entire Payments Settings for, so it is probably out-of-scope for this PR.

https://user-images.githubusercontent.com/3147296/189325927-9d2c3522-8ca2-4334-a726-eb6ff95dea94.mov

**Comment by shendy-a8c** [comment]
That is a good point, @Jinksi! I think it's important to address and it's out of scope for this PR. I quickly searched if there is existing issue to address this problem (disabling the entire settings page when saving) but found none. Could you please create a new issue?

**Comment by Jinksi** [comment]
> Could you please create a new issue?

Thanks Shendy, #4740 created (I couldn't find an existing issue either).

**Comment by aprea** [comment]
p.s. @shendy-a8c, this PR is ready for review.

File: client/settings/deposits/index.js [hunk 1]
```diff
@@ -72,6 +75,17 @@ const monthlyAnchors = [
 ];
 
 const CustomizeDepositSchedule = () => {
+	const {
+		accountStatus: {
+			deposits: { minimum_deposit_amounts: minimumDepositAmounts },
+		},
+		storeCurrencies,
+	} = useContext( WCPaySettingsContext );
```
Comment (shendy-a8c) [inline client/settings/deposits/index.js:83]: If for some reason (maybe unexpected future change from server), `accountStatus.deposits` isn't defined, this line will throw an error, right?

Maybe we can be more defensive here?

**Review by shendy-a8c — CHANGES_REQUESTED** [review]
When I choose Manual, save, and reload the page, the selection becomes Automatic daily. That is not expected right?

https://user-images.githubusercontent.com/73803630/196812104-162ef7fc-bafe-493d-9d91-4bd1b1efbfaf.mov

**Comment by shendy-a8c** [comment]
Also, `ExternalLink` looks weird. It adds `(opens in a new tab)`. No space prefix makes it look weird but it's out of the scope of this PR.

Screen Shot 2022-10-20 at 05 00 50

**Comment by aprea** [comment]
> Also, `ExternalLink` looks weird. It adds `(opens in a new tab)`. No space prefix makes it look weird but it's out of the scope of this PR.

This issue will be handled in https://github.com/Automattic/woocommerce-payments/issues/4015.

File: client/settings/deposits/index.js (hunk 1 above)
Comment (aprea) [reply client/settings/deposits/index.js:83]: I don't think we can be too defensive about the server not returning the expected response. If we need to eventually move/change the payload then it should be done in a backwards-compatible way.

**Comment by aprea** [comment]
> When I choose Manual, save, and reload the page, the selection becomes Automatic daily. That is not expected right?

Good catch! This is because WCPay Server returns the `daily` schedule even if the actual schedule in Stripe is set to `manual`. This happens when we determine that the merchant **has not** completed the new account waiting period. Related code → https://github.com/Automattic/woocommerce-payments-server/blob/5534876318a40f980f1e8d3521b48cc81a726f9b/server/wp-content/rest-api-plugins/endpoints/wcpay/stripe/class-stripe-account.php#L299-L324

I have a PR that is waiting to be reviewed which should fix this issue, would you be able to take a look at it? https://github.com/Automattic/woocommerce-payments-server/pull/2495

**Comment by shendy-a8c** [comment]
Thank you for the explanation.

After testing with https://github.com/Automattic/woocommerce-payments-server/pull/2495, every thing works as expected, , so approving this PR.

I checked:
- Deposit schedule changes from WooCommerce Payments settings page are reflected to the deposit schedule on Stripe dashboard.
- Helper text looks good.
- The deposits settings UI aligns with the latest design update from paJDYF-6a4-p2, ie the dropdowns are disabled instead of hidden.
- Code diff looks good too.

**Review by shendy-a8c — APPROVED** [review]

**Comment by shendy-a8c** [comment]
PR of https://github.com/Automattic/woocommerce-payments/issues/4015, https://github.com/Automattic/woocommerce-payments/pull/4033, seems stale even though it looks ready to go.

**Comment by aheckler** [comment]
@aprea @shendy-a8c The docs URL used by `const depositSettingsDocsUrl` is not ideal. It should be:

`https://woocommerce.com/document/payments/faq/change-deposit-schedule/`

**Comment by aprea** [comment]
Thanks for raising this @aheckler, I'll raise an issue/PR to update that URL.

**Pull request merged**